addons/mail/static/src/js/emojis_mixin.js

```diff
@@ -25,31 +24,19 @@ return {
      * This method should be bound to a click event on an emoji.
      * (used in textarea's emojis dropdown list)
      *
-     * It assumes that a '_getTargetTextArea' method is defined that will return the related
-     * $textarea element.
-     *
-     * Some code from 'basic_composer.js' was copy/pasted to handle the cursor position in the
-     * textarea and handle text replacement if text was selected by the user.
+     * It assumes that a ``_getTargetTextElement`` method is defined that will return the related
+     * textarea/input element in which the emoji will be inserted.
      *
      * @param {MouseEvent} ev
      */
     _onEmojiClick: function (ev) {
-        ev.preventDefault();
-
-        var $target = $(ev.currentTarget);
-        var $textarea = this._getTargetTextArea($target);
+        var unicode = event.target.dataset.emojiUnicode;
+        var textInput = this._getTargetTextElement($(ev.currentTarget))[0];
+        var selectionStart = textInput.selectionStart;
 
-        // copy/pasted from 'basic_composer.js'
-        var inputElement = $textarea.get(0);
-        var cursorPosition = inputElement ? dom.getSelectionRange(inputElement) : {start: 0, end: 0};
-        var inputVal = $textarea.val();
-        var leftSubstring = inputVal.substring(0, cursorPosition.start);
-        var rightSubstring = inputVal.substring(cursorPosition.end);
-        var newInputVal  = [leftSubstring , $target.text().trim(), rightSubstring].join("");
-        var newCursorPosition = newInputVal.length - rightSubstring.length;
-        $textarea.val(newInputVal);
-        $textarea.focus();
-        $textarea[0].setSelectionRange(newCursorPosition, newCursorPosition);
+        textInput.value = textInput.value.slice(0, selectionStart) + unicode + textInput.value.slice(selectionStart);
```

Comment:
Why use plain Javascript instead of jQuery (`.value` instead of `.val()` etc...)?
I know it's working just fine with plain Javascript, but let's keep the jQuery way for coherence with other files.

Reply:
@awa-odoo most of the old code also use vanilla javascript...
For example, with this line we convert a JQuery dom element into a simple javascript dom element, and then we use vanilla methods
```
var inputElement = $textarea.get(0);
```

So, ``inputElement`` is not a JQuery element. The method ``setSelectionRange`` doesn't exist in JQuery.

The only line which use JQuery in the old code is ``$textarea.val();``,  so is it relevant to use JQuery here ?

Reply:
That's a valid point, I guess you can keep the vanilla javascript for this one time.